### ISSUE TITLE
McRaptorSuboptimal ignores later trip patterns

### ISSUE THREAD

**Issue by hbruch** [conveyal/r5#499]
## Expected behavior
McRaptorSuboptimalPathProfileRouter.getPaths() should return trips during the complete time range in between fromTime and toTime.

## Observed behavior
Trips at later departureTimes are ignored, as their states [are pruned](https://github.com/conveyal/r5/blob/dev/src/main/java/com/conveyal/r5/profile/McRaptorSuboptimalPathProfileRouter.java#L577) because their stops were accessed by an earlier departure, e.g. by WALK.

This results in less trip variance and larger average waitTimes in modeify queries.

## Solution idea
Inverting iteration order from incrementing to decrementing departureTimes when [performing each monteCarloRun](https://github.com/conveyal/r5/blob/dev/src/main/java/com/conveyal/r5/profile/McRaptorSuboptimalPathProfileRouter.java#L138) solved this issue for me.

**Comment by ansoncfit** [comment]
I have only used this router for one-to-many travel time results in the travel time calculator (i.e. specifying a non-null InRoutingFareCalculator.Collater and positive maxFare). On first glance, it seems that switching the iteration order would be fine for the one-to-many use case (the order of timesAtStopsEachIteration doesn't matter). Thoughts, @mattwigway ?

**Comment by mattwigway** [comment]
I don't think that switching the iteration order will solve this. I think this is a bug I introduced (sorry!) when I was originally writing this code. I think that around line 146 of McRaptorSuboptimalPathProfileRouter the line `bestTimesAtTargetsByAccessMode.clear()` needs to get called, so there is a separate search state for every search. @hbruch's workaround works for the same reason range-RAPTOR works, but I think the actual bug is much simpler in that I just forgot to include this line.
